core/tests/protractor/fallbacks.js

```diff
@@ -0,0 +1,90 @@
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview End-to-end tests of the fallbacks functionality.
+ *
+ * @author Sean Lip (sean@seanlip.org)
+ */
+
+var general = require('../protractor_utils/general.js');
+var forms = require('../protractor_utils/forms.js');
+var users = require('../protractor_utils/users.js');
+var workflow = require('../protractor_utils/workflow.js');
+var editor = require('../protractor_utils/editor.js');
+var player = require('../protractor_utils/player.js');
+
+describe('Fallbacks editor', function() {
+  it('should allow a fallback to be set and apply it correctly, and ' +
+     'disallow editing of fallbacks in read-only explorations', function() {
+    users.createUser('fallback@example.com', 'fallback');
+    users.login('fallback@example.com');
+
+    workflow.createExploration('sums', 'maths');
+    general.getExplorationIdFromEditor().then(function(explorationId) {
+      editor.enableFallbacks();
+
+      editor.setStateName('card 1');
+      editor.setContent(forms.toRichText('this is card 1'));
+      editor.setInteraction('NumericInput');
+      editor.addResponse('NumericInput', null, 'card 2', true, 'Equals', 21);
+      editor.setDefaultOutcome(forms.toRichText('try again'), null, false);
+      editor.addFallback(
+        2, forms.toRichText('fallback triggered'), 'final card', true);
+
+      editor.moveToState('card 2');
+      editor.setInteraction('Continue', 'click here');
+      editor.setDefaultOutcome(null, 'final card', false);
+
+      editor.moveToState('final card');
+      editor.setInteraction('EndExploration');
+
+      editor.setObjective('To test fallbacks.');
+      editor.saveChanges();
+      workflow.publishExploration();
+
+      // Login as another user and verify that the exploration editor does not
+      // allow the second user to modify the exploration.
+      users.logout();
+      users.login('user7@example.com');
+      general.openEditor(explorationId);
+      editor.exitTutorialIfNecessary();
+
+      // Verify nothing can change with this user.
+      editor.expectInteractionToMatch('NumericInput');
+      editor.expectCannotAddFallback();
+
+      var fallbackEditor = editor.FallbackEditor(0);
+      fallbackEditor.expectCannotSetFeedback();
+      fallbackEditor.expectCannotSetDestination();
+      fallbackEditor.expectCannotDeleteFallback();
+      fallbackEditor.expectCannotChangeTriggerCondition();
+
+      general.moveToPlayer();
+      player.submitAnswer('NumericInput', 19);
+      player.expectLatestFeedbackToMatch(forms.toRichText('try again'));
+      player.expectExplorationToNotBeOver();
+      player.submitAnswer('NumericInput', 18);
+      player.expectLatestFeedbackToMatch(forms.toRichText('fallback triggered'));
```

Comment:
I would add a check for rules that go somewhere taking precedence over a fallback that was about to trigger.